### PR TITLE
api: add dclocation in member response

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: Song Gao <disxiaofei@163.com>

### What is changed and how it works?

add dclocation in member response

ref https://github.com/pingcap/kvproto/pull/721

### Check List

Tests

- Unit test

### Release note

* No release note

### CODE REVIEW

File: server/api/member.go [hunk 1]
```diff
@@ -64,7 +64,12 @@ func getMembers(svr *server.Server) (*pdpb.GetMembersResponse, error) {
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
+	dclocationDistribution, err := svr.GetTSOAllocatorManager().GetClusterDCLocationsFromEtcd()
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
 	for _, m := range members.GetMembers() {
+		m.DcLocation = "global"
```
Comment (JmPotato) [inline server/api/member.go:72]: Does it mean this filed is `global` by default? What if we have a PD that doesn't configure `dc-location`? Should it be `global` also?

Comment (Yisaer) [reply server/api/member.go:72]: Yes.

File: go.mod [hunk 2]
```diff
@@ -4,7 +4,7 @@ go 1.13
 
 require (
 	github.com/BurntSushi/toml v0.3.1
-	github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751
+	github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751 // indirect
```
Comment (JmPotato) [inline go.mod:7]: Is there something wrong with this dependency? Why does it change?

File: server/api/member.go [hunk 3]
```diff
@@ -91,6 +96,19 @@ func getMembers(svr *server.Server) (*pdpb.GetMembersResponse, error) {
...
+		found := false
+		for dcLocation, serverIDs := range dclocationDistribution {
+			for _, serverID := range serverIDs {
+				if serverID == m.MemberId {
+					m.DcLocation = dcLocation
+					found = true
+					break
+				}
+			}
+			if found {
```
Comment (rleungx) [inline server/api/member.go:108]: What is the `found` used for? how about directly `return members, nil` if `serverID == m.MemberId`?

Comment (Yisaer) [reply server/api/member.go:108]: I think we can't. We should fulfill `dcLocation` for every member.

Comment (rleungx) [reply server/api/member.go:108]: oh, my mistake~

File: server/api/member_test.go [hunk 4]
```diff
@@ -38,7 +38,12 @@ type testMemberAPISuite struct {
 }
 
 func (s *testMemberAPISuite) SetUpSuite(c *C) {
-	s.cfgs, s.servers, s.clean = mustNewCluster(c, 3)
+	s.cfgs, s.servers, s.clean = mustNewCluster(c, 3, func(cfg *config.Config) {
+		cfg.EnableLocalTSO = true
+		cfg.Labels = map[string]string{
+			"zone": "dc-1",
```
Comment (JmPotato) [inline server/api/member_test.go:44]: Better use `config.ZoneLabel` here.

Comment (Yisaer) [reply server/api/member_test.go]: addressed.

**Review by JmPotato — APPROVED** [review]

File: server/api/member.go (hunk 1 above)
Comment (nolouch) [reply server/api/member.go:72]: How about empty if not config the zone label?

Comment (nolouch) [reply server/api/member.go:72]: "Global" is a concept in the tso allocator, and has no strong correlation with which DC localted.

Comment (Yisaer) [reply server/api/member.go:72]: good point.

**Review by rleungx — APPROVED** [review]

**Review by rleungx — APPROVED** [review]

**Pull request merged**